### PR TITLE
Fix expiring tile caches by giving PutBucketLifecycleConfiguration permissions to data API

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

- The Data API didn't have sufficient privileges to change the lifecycle configuration on the tile cache bucket (which is how one mass-expires things), thus fails to clean-up tile caches for deleted versions and assets

Issue Number: GTC-2708

## What is the new behavior?

- The Data API is given sufficient privileges to change the lifecycle configuration on the tile cache bucket, and can now delete tile caches.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: terraform/templates/tile_cache_bucket_policy.json.tmpl [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListBucket",
+                "s3:PutLifecycleConfiguration"
```
Comment (danscales) [inline terraform/templates/tile_cache_bucket_policy.json.tmpl:8]: This should be:

"s3:PutBucketLifecycleConfiguration"

**Review by danscales — COMMENTED** [review]
Need to fix the permission name (see my comment).

Also, will be good to see the tests run completely once they can access the new gfw-tile-cache definitions.

Thanks for doing this!

File: terraform/templates/tile_cache_bucket_policy.json.tmpl (hunk 1 above)
Comment (dmannarino) [reply terraform/templates/tile_cache_bucket_policy.json.tmpl:8]: Good catch, thanks!
I wonder why it didn't complain?

File: terraform/main.tf [hunk 2]
```diff
@@ -105,7 +105,7 @@ module "fargate_autoscaling" {
     aws_iam_policy.iam_api_gateway_policy.arn,
     aws_iam_policy.read_gcs_secret.arn,
     data.terraform_remote_state.tile_cache.outputs.ecs_update_service_policy_arn,
-    data.terraform_remote_state.tile_cache.outputs.tile_cache_bucket_full_access_policy_arn,
+    aws_iam_policy.tile_cache_bucket_policy.arn,
```
Comment (gtempus) [inline terraform/main.tf:108]: I like the move towards keeping the permission close to the project that needs it (data-api). 💯

**Review by gtempus — APPROVED** [review]
I walked through this with @dmannarino during a pairing session. Thanks, Daniel!

**Pull request merged**